File: 746. Min Cost Climbing Stairs/main.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int minCostClimbingStairs(vector<int> &cost) {
        int costSize = cost.size();
        vector<int> dp(costSize + 1, 0);

        for (int i = 2; i <= costSize; i++) {
            dp[i] = min(dp[i - 2] + cost[i - 2], dp[i - 1] + cost[i - 1]);
        }

        return dp[costSize];
    }
};

File: 746. Min Cost Climbing Stairs/main.rs
struct Solution;

impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let cost_size = cost.len();
        let mut dp = vec![0; cost_size + 1];

        for i in 2..=cost_size {
            dp[i] = (dp[i - 2] + cost[i - 2]).min(dp[i - 1] + cost[i - 1]);
        }

        dp[cost_size]
    }
}
